### PR TITLE
Added measures to avoid performance hungy asssert handling

### PR DESCRIPTION
**Pull request by weissm**
## IS YOUR COMMIT SIGNED?
Note that this repo requires all commits to be properly signed, [see here](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) for more info

## Enhancement or Defect Addressed by This PR
Enhancement 

## Description of Proposed Changes
Added additional condition to check saved stated to avoid later assert which is very performance hungry

### CODE REVIEW

File: OneMore/Commands/Tagging/HashtagService.cs [hunk 1]
```diff
@@ -211,6 +211,9 @@ protected async Task Scan()
 			if (notebookFilters is not null && notebookFilters.Length > 0)
 			{
 				scanner.SetNotebookFilters(notebookFilters);
+			} else
```
Comment (stevencohn) [inline OneMore/Commands/Tagging/HashtagService.cs:214]: This ignores and breaks the "all" notebooks condition. While the UI currently might avoid this case, it was an intentional and possible future case. I want to leave it in.

File: OneMore/Commands/Tagging/HashtagScanner.cs [hunk 2]
```diff
@@ -380,7 +380,9 @@ private async Task<bool> ScanPage(
 
 			// saved tags will be in document-order but not have DocumentOrder set,
 			// we can rely on tag + objectID to continue resolving
-			var saved = provider.ReadPageTags(pageID);
+
+			// in case of empty candidate list do not read page tags as it throws exception
+			var saved = (candidates.Count() == 0) ? null : provider.ReadPageTags(pageID);
```
Comment (stevencohn) [inline OneMore/Commands/Tagging/HashtagScanner.cs:385]: This break the case where there are no longer hashtags on the page but there used to be, and now the scanner will not delete the old hashtags in the db.

**Review by stevencohn — CHANGES_REQUESTED** [review]
Where are the asserts to which you refer? Internals to the SQLite library?

File: OneMore/Commands/Tagging/HashtagScanner.cs [hunk 3]
```diff
@@ -380,7 +380,9 @@ private async Task<bool> ScanPage(
 
 			// saved tags will be in document-order but not have DocumentOrder set,
 			// we can rely on tag + objectID to continue resolving
-			var saved = provider.ReadPageTags(pageID);
+
+			// in case of empty candidate list do not read page tags as it throws exception
```
Comment (stevencohn) [inline OneMore/Commands/Tagging/HashtagScanner.cs:384]: I haven't been able to reproduce this case. How would it happen?

File: OneMore/Commands/Tagging/HashtagService.cs (hunk 1 above)
Comment (stevencohn) [reply OneMore/Commands/Tagging/HashtagService.cs:214]: ping

**Review by stevencohn — CHANGES_REQUESTED** [review]

**Comment by weissm** [comment]
I also deep dive into this problem. What appearntly happend on my side was that the sql file "OneMore.db" was corrupted. With this the assert was multiple times called, caused to slow down my onenote system. After removing the file it was newly created and apperently did not show this behavior again. Not clear if there is a chance to double check the consistancy of OneMore.db. Otherwise, this issue could be closed.

**Pull request closed without merging**